src/groupeddataframe/groupeddataframe.jl

```diff
@@ -323,6 +323,14 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function Base.to_index(gd::GroupedDataFrame, key::Tuple{Vararg{<:Pair}})
+    println("hey there")
+    if length(key) != length(gd.cols) || any(n != _names(gd)[c] for (n, c) in zip(Symbol.(first.(key)), gd.cols))
+        throw(KeyError(key))
+    end
+    return Base.to_index(gd, ntuple(i -> last(key[i]), length(key)))
```

Comment:
this seems to have the same performance and reads a bit cleaner to me:
```suggestion
    return Base.to_index(gd, last.(key))
```